pkg/tls/reader.go

```diff
@@ -48,11 +66,28 @@ func ReadTLSPair(restConfig *rest.Config, client *client.Client) (*PemPair, erro
 		return nil, errors.Wrap(err, "failed to get TLS Cert Properties")
 	}
 
+	depl, err := client.GetResource("", "Deployment", certProps.Namespace, config.KyvernoDeploymentName)
+
+	deplHash := ""
+	if err == nil {
+		deplHash = fmt.Sprintf("%v", depl.GetUID())
+	}
+
+	var deplHashSec string = "default"
+	var ok, managedByKyverno bool
+
 	sname := generateTLSPairSecretName(certProps)
 	unstrSecret, err := client.GetResource("", "Secret", certProps.Namespace, sname)
 	if err != nil {
 		return nil, fmt.Errorf("failed to get secret %s/%s: %v", certProps.Namespace, sname, err)
 	}
+	if label, ok := unstrSecret.GetLabels()[ManagedByLabel]; ok {
+		managedByKyverno = label == "kyverno"
+	}
+	deplHashSec, ok = unstrSecret.GetAnnotations()[MasterDeploymentUID]
+	if managedByKyverno && (!ok || deplHashSec != deplHash) {
+		return nil, fmt.Errorf("outdated secret")
```

Comment:
`ReadTLSPair` is used in multiple places, is it safe to return the error if the secret is outdated? Do we have corresponding logic to re-create the secret for all references?
![image](https://user-images.githubusercontent.com/25727662/148261574-da910a3c-e856-46b2-aab6-3be2c9f141ce.png)

Reply:
Wouldn't it be safer to return the error in case the Secret is found to be stale?

Reply:
Take the third reference as an example, it is used in `pkg/webhookconfig` `retryReadTLS`, and Kyverno will eventually panic if `GetTLSPemPair ` returns an error:
https://github.com/kyverno/kyverno/blob/e9826e103dc8aaeb054ef6a71c5ea2e256462753/cmd/kyverno/main.go#L437-L441
How can we recover from this scenario?

Reply:
Makes sense. Maybe we can check for the error we are returning separately and just log it in that case?

Reply:
Log out the error can't solve the issue, Kyverno won't start if the secret is outdated. We need to re-create the certs and update the secret if its not managed by the current instance.

Reply:
`GetTLSPemPair()` repeatedly checks if `ReadTLSPair()` is returning an error while `CertManager` is running concurrently to generate/load the Secrets right? When the Pod starts, it invokes `CertRenewer.InitTLSPemPair()` which checks if the ReadTLSPair() returns an error otherwise it creates a new Secret.

Reply:
Will there be any race condition?

Reply:
IDTS. It's the same behaviour as we currently have for missing Secret (as on line 82 here), we just add a check for stale.

Reply:
Ok. Can you address other comments so we could proceed?

Reply:
Sure